src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -1699,14 +1699,14 @@ export default abstract class ComparisonStore extends AnalysisStore
     readonly gaEnrichmentGroupsByAssayType = remoteData({
         await: () => [
             this
-                .selectedGenericAssayEnrichmentProfileMapGroupedByGenericAssayType,
+                .selectedAllGenericAssayEnrichmentProfileMapGroupedByGenericAssayType,
```

Comment:
Just wondering, after your changes do we still need the following `action` and `readonly` :
- `action setGenericAssayEnrichmentProfileMap`  (line 1095)
-  `readonly selectedGenericAssayEnrichmentProfileMapGroupedByGenericAssayType` (line 914)

It looks like `selectedGenericAssayEnrichmentProfileMapGroupedByGenericAssayType` was only used in the functions where you made the changes. If they are not needed anymore, might be nice to remove them from the code (to clean up)

Reply:
I remember that the `selectedGenericAssayEnrichmentProfileMapGroupedByGenericAssayType` is used by the UI to determine which profile is currently being displayed. Maybe @MatthijsPon can double-check to confirm if it is necessary or not.

Reply:
@dippindots The only place where I can find `selectedGenericAssayEnrichmentProfileMapGroupedByGenericAssayType` being used, is in the file [src/pages/groupComparison/GenericAssayEnrichments.tsx](https://github.com/thehyve/cbioportal-frontend/blob/fix_ga_profile_switch/src/pages/groupComparison/GenericAssayEnrichments.tsx). This file seems to have been made obsolete by [src/pages/groupComparison/GenericAssayEnrichmentCollections.tsx](https://github.com/thehyve/cbioportal-frontend/blob/fix_ga_profile_switch/src/pages/groupComparison/GenericAssayEnrichmentCollections.tsx)?